### PR TITLE
CI: Compare API compatibility against both 4.0 and 4.1

### PR DESCRIPTION
**Pull request by RedworkDE**
Allows running the API compat checks against multiple Godot versions without having to duplicate a bunch of entries by using a scheme were, if the basename of an expected file contains an `_` it will be compared against the version indicated by the part before the `_` but the expected errors will be concatenated to those of the version after the underscore (possibly recursively).

This changes prevents accidentally breaking APIs that there newly added in 4.1.

### CODE REVIEW

**Comment by dsnopek** [comment]
This approach seems good to me!

**Review by akien-mga — APPROVED** [review]
Looks good to me. Needs a rebase.

File: misc/extension_api_validation/4.1-stable.expected [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Validate extension JSON: API was removed: classes/GodotPhysicsServer2D
+Validate extension JSON: API was removed: classes/GodotPhysicsServer3D
+Validate extension JSON: API was removed: classes/IPUnix
+Validate extension JSON: API was removed: classes/MovieWriterMJPEG
+Validate extension JSON: API was removed: classes/MovieWriterPNGWAV
+Validate extension JSON: API was removed: classes/ResourceFormatImporterSaver
+Validate extension JSON: API was removed: classes/FramebufferCacheRD
+Validate extension JSON: API was removed: classes/UniformSetCacheRD
+Validate extension JSON: API was removed: classes/GLTFDocumentExtensionPhysics
+Validate extension JSON: API was removed: classes/GLTFDocumentExtensionTextureWebP
```
Comment (RedworkDE) [inline misc/extension_api_validation/4.1-stable.expected:116]: These two entries are new, but they are mentioned in this PR as being removed from the export, so this is correct and exactly the kind of thing that could slip through before this.

Comment (akien-mga) [reply misc/extension_api_validation/4.1-stable.expected:116]: The GitHub action still raises a warning for those but on the wrong file:
image

Comment (RedworkDE) [reply misc/extension_api_validation/4.1-stable.expected:116]: I have hidden warnings from prior versions (as I originally intended), as it is kinda unavoidable that they happen (unless editing the expected files for old versions). (tho thinking about it there may still be some issues when changing a value in multiple versions, potentially still requiring changes to old expected files)

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**